ecommerce/hubspot_api_test.py

```diff
@@ -0,0 +1,89 @@
+"""
+Hubspot API tests
+"""
+# pylint: disable=redefined-outer-name
+from urllib.parse import urlencode, urljoin
+
+import pytest
+from faker import Faker
+from django.conf import settings
+
+from ecommerce.hubspot_api import (
+    send_hubspot_request,
+    HUBSPOT_API_BASE_URL,
+    make_sync_message,
+    make_contact_sync_message,
+)
+from users.serializers import UserSerializer
+
+fake = Faker()
+
+
+@pytest.mark.parametrize("request_method", ["GET", "PUT", "POST"])
+def test_send_hubspot_get_request(mocker, request_method):
+    """Test sending hubspot request with method = GET"""
+    endpoint = fake.uri_page()
+    api_url = fake.uri_path()
+    value = fake.pyint()
+    query_params = {"param": value}
+
+    # Include hapikey when generating url to match request call against
+    full_query_params = {"param": value, "hapikey": settings.HUBSPOT_API_KEY}
+    mock_request = mocker.patch(
+        f"ecommerce.hubspot_api.requests.{request_method.lower()}"
+    )
+    url_params = urlencode(full_query_params)
+    base_url = urljoin(HUBSPOT_API_BASE_URL, api_url)
+    url = f"{urljoin(base_url, endpoint)}?{url_params}"
+    if request_method == "GET":
+        send_hubspot_request(endpoint, api_url, "GET", query_params=query_params)
+        mock_request.assert_called_once_with(url=url)
+    else:
+        body = fake.pydict()
+        if request_method == "PUT":
```

Comment:
How about something like this with real API url's and without the `elif` block :
```python
@pytest.mark.parametrize("request_method", ["GET", "PUT", "POST"])
@pytest.mark.parametrize(
    "endpoint,api_url,expected_url",
    [
        [
            "sync-errors",
            "/extensions/ecomm/v1",
            f"{HUBSPOT_API_BASE_URL}/extensions/ecomm/v1/sync-errors",
        ],
        [
            "",
            "/extensions/ecomm/v1/installs",
            f"{HUBSPOT_API_BASE_URL}/extensions/ecomm/v1/installs",
        ],
        [
            "CONTACT",
            "/extensions/ecomm/v1/sync-messages",
            f"{HUBSPOT_API_BASE_URL}/extensions/ecomm/v1/sync-messages/CONTACT",
        ],
    ],
)
def test_send_hubspot_request(mocker, request_method, endpoint, api_url, expected_url):
    """Test sending hubspot request with method = GET"""
    value = fake.pyint()
    query_params = {"param": value}

    # Include hapikey when generating url to match request call against
    full_query_params = {"param": value, "hapikey": settings.HUBSPOT_API_KEY}
    mock_request = mocker.patch(
        f"ecommerce.hubspot_api.requests.{request_method.lower()}"
    )
    url_params = urlencode(full_query_params)
    url = f"{expected_url}?{url_params}"
    if request_method == "GET":
        send_hubspot_request(
            endpoint, api_url, request_method, query_params=query_params
        )
        mock_request.assert_called_once_with(url=url)
    else:
        body = fake.pydict()
        send_hubspot_request(
            endpoint, api_url, request_method, query_params=query_params, body=body
        )
        mock_request.assert_called_once_with(url=url, json=body)
```

Reply:
Just saw your latest commit, take or leave the above suggestion as long as tests pass.